ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -236,13 +238,11 @@ public org.omg.CosTransactions.Vote prepare() throws HeuristicMixed,
 			 * XAER_PROTO.
 			 */
 
-			if (_rollbackOptimization) // won't have rollback called on it
-				removeConnection();
```

Comment:
@jmfinelli it's essentially the same question as I had above but for this other class and so the control flow might be slightly different. So for this class when`_rolledBack = true` during topLevelPrepare() are you sure that removeConnection() is called (in topLevelAbort() somewhere perhaps)?

Reply:
Thank you!